pynecone/state.py

```diff
@@ -706,32 +703,24 @@ def get_delta(self) -> Delta:
         if len(subdelta) > 0:
             delta[self.get_full_name()] = subdelta
 
-        # Recursively find the substate deltas.
-        substates = self.substates
-        for substate in self.dirty_substates:
-            delta.update(substates[substate].get_delta())
+        # Recursively find all substate deltas.
+        for substate in self.substates.values():
```

Comment:
I think we may want to keep the `dirty_substates` prop. This will work, but as the substate tree grows very big, this may end up having some performance impact as we will call `get_delta()` for every substate in the tree.